src/app/calculator/furnaces/o2-enrichment/enrichment-graph/enrichment-graph.component.ts

```diff
@@ -0,0 +1,381 @@
+    this.selectedDataPoints = this.o2EnrichmentService.selectedDataPoints.getValue();
+  }
+
+  updateChart() {
+    this.updateTraces();
+    let chartLayout = JSON.parse(JSON.stringify(this.enrichmentChart.layout));
+    Plotly.update(this.currentChartId, this.enrichmentChart.data, chartLayout);
+    this.save();
+  }
+
+  setTraces() {
+    this.enrichmentInputs.forEach((enrichmentInput: EnrichmentInput, index) => {
+      // Line trace
+      let graphData: TraceCoordinates = this.o2EnrichmentService.getGraphData(this.settings, enrichmentInput.inputData);
+      let lineTrace = this.o2EnrichmentService.getLineTrace();
+      lineTrace.x = graphData.x;
+      lineTrace.y = graphData.y;
+      lineTrace.line.color = this.getNextColor();
+      
+      // Point trace
+      let outputs = this.o2EnrichmentService.enrichmentOutputs.getValue();
+      let fuelSavings = outputs[index].outputData.fuelSavings;
+      let displayPoint: DisplayPoint = {
+        name: enrichmentInput.inputData.name,
+        pointColor: this.getNextColor(),
+        pointX: enrichmentInput.inputData.o2CombAir,
+        pointY: fuelSavings,
+        combustionTemp: enrichmentInput.inputData.combAirTemp,
+        flueOxygen: enrichmentInput.inputData.o2FlueGas,
+        fuelTemp: enrichmentInput.inputData.flueGasTemp
+      }
+      let pointTrace = this.o2EnrichmentService.getPointTrace(displayPoint);
+      pointTrace.marker.color = this.getNextColor();
+      pointTrace.marker.line.color = this.getNextColor();
+
+      if (index == 0) {
+        lineTrace.line.color = "#000";
+        pointTrace.marker.color = this.defaultPointColor;
+        pointTrace.marker.line.color = this.defaultPointOutlineColor;
+        displayPoint.pointColor = "#000";
+      }
+      
+      this.enrichmentChart.data.push(lineTrace, pointTrace);
+      this.selectedDataPoints.push(displayPoint);
+    });
+    this.enrichmentChart.inputCount = this.enrichmentInputs.length;
+  }
+
+  updateTraces() {
+    let addingTraces: boolean = this.enrichmentInputs.length > this.enrichmentChart.inputCount;
+    let removingTraces: boolean = this.enrichmentInputs.length < this.enrichmentChart.inputCount;
+    if (removingTraces) {
+      // Remove line
+      this.enrichmentChart.data.splice(this.enrichmentChart.data.length - 2, 2);
+      // Remove point
+      this.selectedDataPoints.splice(this.selectedDataPoints.length -1, 1);
+    }
+    this.enrichmentInputs.forEach((enrichmentInput: EnrichmentInput, index) => {
+      let isNewTrace: boolean = false;
+      if (addingTraces) {
+        let offset = this.enrichmentInputs.length - this.enrichmentChart.inputCount;
+        let newTraceCount = Array.from(Array(offset).keys())
+        if (newTraceCount.includes(index - 1)) {
+          isNewTrace = true;
+        }
+      }
+      let lineIndex = index + index;
+      let pointIndex = lineIndex + 1;
+      let lineTraceColor;
```

Comment:
Same color comment as above, more calls to getNextColor() and variables then needed